Wrong argument to `Path` in ipython in version 8.0.0a1

<!-- This is the repository for IPython command line, if you can try to make sure this question/bug/feature belong here and not on one of the Jupyter repositories. 

If it's a generic Python/Jupyter question, try other forums or discourse.jupyter.org.

If you are unsure, it's ok to post here, though, there are few maintainer so you might not get a fast response. 

-->

I have pre-release tests on my package and I got such error (only on windows  https://github.com/4DNucleome/PartSeg/runs/4559230495?check_suite_focus=true) :

```
  D:\tmp\tox\py38-PyQt5-all\lib\site-packages\PartSeg\common_gui\qt_console.py:9: in <module>
      from ipykernel.inprocess.ipkernel import InProcessInteractiveShell
  D:\tmp\tox\py38-PyQt5-all\lib\site-packages\ipykernel\inprocess\ipkernel.py:10: in <module>
      from IPython.core.interactiveshell import InteractiveShellABC
  D:\tmp\tox\py38-PyQt5-all\lib\site-packages\IPython\__init__.py:51: in <module>
      from .core.application import Application
  D:\tmp\tox\py38-PyQt5-all\lib\site-packages\IPython\core\application.py:37: in <module>
      programdata = Path(os.environ.get("PROGRAMDATA", None))
  c:\hostedtoolcache\windows\python\3.8.10\x64\lib\pathlib.py:1042: in __new__
      self = cls._from_parts(args, init=False)
  c:\hostedtoolcache\windows\python\3.8.10\x64\lib\pathlib.py:683: in _from_parts
      drv, root, parts = self._parse_args(args)
  c:\hostedtoolcache\windows\python\3.8.10\x64\lib\pathlib.py:667: in _parse_args
      a = os.fspath(a)
  E   TypeError: expected str, bytes or os.PathLike object, not NoneType
  ERROR: InvocationError for command 'D:\tmp\tox\py38-PyQt5-all\Scripts\pytest.EXE' package/tests --no-cov (exited with code 4)
  ```

It looks like wrong default value is provided here `os.environ.get("PROGRAMDATA", None)`

--- a/IPython/core/application.py
+++ b/IPython/core/application.py
@@ -34,9 +34,9 @@ from traitlets import (
 )
 
 if os.name == "nt":
-    programdata = Path(os.environ.get("PROGRAMDATA", None))
-    if programdata:
-        SYSTEM_CONFIG_DIRS = [str(programdata / "ipython")]
+    programdata = os.environ.get("PROGRAMDATA", None)
+    if programdata is not None:
+        SYSTEM_CONFIG_DIRS = [str(Path(programdata) / "ipython")]
     else:  # PROGRAMDATA is not defined by default on XP.
         SYSTEM_CONFIG_DIRS = []
 else:

